spec/rdiff-backup-test/server_spec.rb

```diff
@@ -0,0 +1,163 @@
+    context "on #{pltfrm[:platform]} #{pltfrm[:version]}" do
+      let(:runner) do
+        ChefSpec::SoloRunner.new(
+        pltfrm.dup.merge(step_into: ['rdiff-backup'])
+        )
+      end
+      cached(:chef_run) { runner.converge(described_recipe) }
+
+        before do
+          stub_command('secrets').and_return(false)
+          Chef::EncryptedDataBagItem.stub(:load).with('rdiff-backup-secrets','secrets').and_return({
+              key: 'secret-key'
+          })
+        end
+        it do
+          expect(chef_run).to include_recipe('yum-epel')
+        end
+        %w(rdiff-backup cronolog).each do |r|
+          it do
+            expect(chef_run).to install_package(r)
+          end
+        end
+
+        it do
+          expect(chef_run).to include_recipe('nrpe')
+        end
+        it do
+          expect(chef_run).to create_cookbook_file(
+          chef_run.node['nrpe']['plugin_dir'] + '/check_rdiff').with(
+            mode: 0755,
+            cookbook: 'rdiff-backup',
+            owner: chef_run.node['nrpe']['user'],
+            group: chef_run.node['nrpe']['group'],
+            source: 'nagios/plugins/check_rdiff',
+            cookbook: 'rdiff-backup'
+          )
+        end
+        # it do
+        #   expect(chef_run).to create_file('check_rdiff_job_test1').with(
+        #   command: '/usr/bin/sudo/' + chef_run.node['nrpe']['plugin_dir'] +
+        #            'check_rdiff' + "-w 16 "\
+        #                            "-c 18 "\
+        #                            "-r /data/rdiff-backup "\
+        #                            "-p 24 "\
+        #                            "-l 800000000"
+        #   )
+        # end
+          it do
+            expect(chef_run).to create_user('rdiff-backup-server')
+          end
+          it do
+          expect(chef_run).to install_sudo('rdiff-backup-server').with(
+            user: 'rdiff-backup-server',
+            group: 'rdiff-backup-server',
+            nopasswd: true,
+            commands: ['/usr/bin/sudo rdiff-backup '\
+                      '--server --restrict-read-only /']
+            )
+          end
+          it do
+            expect(chef_run).to create_directory('/var/rdiff-backup/locks').with(
+              owner: 'rdiff-backup-server',
+              group: 'rdiff-backup-server' || 'rdiff-backup-server', # Group || Owner
+              mode: 0755,
+              recursive: true
+            )
+          end
+          it do
+            expect(chef_run).to create_directory('/home/rdiff-backup-server/.ssh').with(
+              owner: 'rdiff-backup-server',
+              group: 'rdiff-backup-server' || 'rdiff-backup-server', # Group || Owner
+              mode: 0700
+            )
+          end
+          # it do
+          #   expect(chef_run).to create_ssh_user('id_rsa').with(
+          #   user: 'rdiff-backup-server'
+          #   key: secrets['ssh-key']
+          #   )
+          # end
```

Comment:
I believe you need to step into the ssh_user_private_key resource to make this work.